packages/expect/src/toThrowMatchers.js

```diff
@@ -25,32 +25,44 @@ const DID_NOT_THROW = 'Received function did not throw';
 
 type Thrown =
   | {
+      hasMessage: true,
       isError: true,
       message: string,
       value: Error,
     }
   | {
+      hasMessage: boolean,
       isError: false,
       message: string,
       value: any,
     };
 
-const getThrown = (e: any): Thrown =>
-  e !== null &&
-  e !== undefined &&
-  typeof e.message === 'string' &&
-  typeof e.name === 'string' &&
-  typeof e.stack === 'string'
-    ? {
-        isError: true,
-        message: e.message,
-        value: e,
-      }
-    : {
-        isError: false,
-        message: typeof e === 'string' ? e : String(e),
-        value: e,
-      };
+const getThrown = (e: any): Thrown => {
+  if (
+    e !== null &&
+    e !== undefined &&
+    typeof e.message === 'string' &&
+    typeof e.name === 'string' &&
+    typeof e.stack === 'string'
+  ) {
+    return {
+      hasMessage: true,
+      isError: true,
+      message: e.message,
+      value: e,
+    };
+  }
+
+  const hasMessage =
+    e !== null && e !== undefined && typeof e.message === 'string';
+
+  return {
+    hasMessage,
+    isError: false,
+    message: hasMessage ? e.message : typeof e === 'string' ? e : String(e),
```

Comment:
Would prefer
```suggestion
    message: hasMessage ? e.message : String(e),
```
for better readability.
I assume `String('already a string')` is probably optimized well enough to not be significantly more expensive than the typeof check anyway

Reply:
it's not hot code anyways, so doesn't matter